monai/data/image_reader.py

```diff
@@ -0,0 +1,348 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+
+from monai.data.utils import correct_nifti_header_if_necessary
+from monai.utils import ensure_tuple, optional_import
+
+itk, _ = optional_import("itk", allow_namespace_pkg=True)
+nib, _ = optional_import("nibabel")
+
+
+class ImageReader(ABC):
+    """Abstract class to define interface APIs to load image files.
+    users need to call `read` to load image and then use `get_data`
+    to get the image data and properties from meta data.
+
+    Args:
+        img: image to initialize the reader, this is for the usage that the image data
+            is already in memory and no need to read from file again, default is None.
+
+    """
+
+    def __init__(self, img: Optional[Any] = None) -> None:
+        self.img = img
+        self._suffixes: Optional[Sequence[str]] = None
+
+    def get_suffixes(self) -> Sequence[str]:
```

Comment:
this is not used anywhere, could we remove it, and add it back if it's really necessary?

Reply:
removed in the latest commit.
Thanks.